crates/miden-testing/tests/auth/multisig.rs

```diff
@@ -0,0 +1,414 @@
+use miden_objects::testing::account_id::{
+    ACCOUNT_ID_PUBLIC_FUNGIBLE_FAUCET,
+    ACCOUNT_ID_REGULAR_PUBLIC_ACCOUNT_UPDATABLE_CODE,
+};
+use miden_objects::transaction::OutputNote;
+use miden_objects::vm::AdviceMap;
+use miden_objects::{Felt, Hasher, Word};
+use miden_testing::{Auth, MockChainBuilder};
+use miden_tx::TransactionExecutorError;
+use miden_tx::auth::{BasicAuthenticator, SigningInputs, TransactionAuthenticator};
+use miden_tx::utils::word_to_masm_push_string;
+use rand::SeedableRng;
+use rand_chacha::ChaCha20Rng;
+use vm_processor::crypto::RpoRandomCoin;
+
+// ================================================================================================
+// HELPER FUNCTIONS
+// ================================================================================================
+
+type MultisigTestSetup = (Vec<SecretKey>, Vec<PublicKey>, Vec<BasicAuthenticator<ChaCha20Rng>>);
+
+/// Sets up secret keys, public keys, and authenticators for multisig testing
+fn setup_keys_and_authenticators(
+    num_approvers: usize,
+    threshold: usize,
+) -> anyhow::Result<MultisigTestSetup> {
+    let mut rng = ChaCha20Rng::from_seed([0u8; 32]);
+
+    let mut secret_keys = Vec::new();
+    let mut public_keys = Vec::new();
+    let mut authenticators = Vec::new();
+
+    for _ in 0..num_approvers {
+        let sec_key = SecretKey::with_rng(&mut rng);
+        let pub_key = sec_key.public_key();
+
+        secret_keys.push(sec_key);
+        public_keys.push(pub_key);
+    }
+
+    // Create authenticators only for the signers we'll actually use
+    for i in 0..threshold {
+        let authenticator = BasicAuthenticator::<ChaCha20Rng>::new_with_rng(
+            &[(public_keys[i].into(), AuthSecretKey::RpoFalcon512(secret_keys[i].clone()))],
+            rng.clone(),
+        );
+        authenticators.push(authenticator);
+    }
+
+    Ok((secret_keys, public_keys, authenticators))
+}
+
+/// Creates a multisig account with the specified configuration
+fn create_multisig_account(
+    threshold: u32,
+    public_keys: &[PublicKey],
+    asset_amount: u64,
+) -> anyhow::Result<Account> {
+    let approvers: Vec<_> = public_keys.iter().map(|pk| (*pk).into()).collect();
+
+    let multisig_account = AccountBuilder::new([0; 32])
+        .with_auth_component(Auth::Multisig { threshold, approvers })
+        .with_component(BasicWallet)
+        .account_type(AccountType::RegularAccountUpdatableCode)
+        .storage_mode(AccountStorageMode::Public)
+        .with_assets(vec![FungibleAsset::mock(asset_amount)])
+        .build_existing()?;
+
+    Ok(multisig_account)
+}
+
+// ================================================================================================
+// TESTS
+// ================================================================================================
+
+#[tokio::test]
+async fn test_multisig() -> anyhow::Result<()> {
+    // ROLES
+    // - 2 Approvers (multisig signers)
+    // - 1 Multisig Contract
```

Comment:
Should we give this a more descriptive name so it gives an idea what it tests and how it is different from the other multisig tests? I would make the roles description the doc comment of the test with some short description. Same for the other tests here - I think it just requires moving the comment above the test and make it a doc comment.